pontos/version/commands/_java.py

```diff
@@ -15,199 +15,160 @@
 
-        if current_version != version:
+        if last_version != str(version):
             raise VersionError(
                 f"Provided version {version} does not match the "
-                f"current version {current_version} in "
-                f"{self.project_file_path}."
+                f"current version {last_version} "
+                f"in '{self.project_file_path}'"
             )
 
     def update_version(
         self, new_version: Version, *, force: bool = False
     ) -> VersionUpdate:
         try:
-            package_version = self.get_current_version()
-            if not force and new_version == package_version:
-                return VersionUpdate(previous=package_version, new=new_version)
+            current_version = self.get_current_version()
+            if not force and new_version == current_version:
+                return VersionUpdate(previous=current_version, new=new_version)
         except VersionError:
             # just ignore current version and override it
-            package_version = None
+            current_version = None
 
-        changed_files = [self.project_file_path]
-        self._update_pom_version(new_version=new_version)
-        self._update_properties_file(new_version=new_version)
-        self._update_swagger_config(new_version=new_version)
+        changed_files = self._update_version_files(new_version)
 
         return VersionUpdate(
-            previous=package_version,
+            previous=current_version,
             new=new_version,
             changed_files=changed_files,
         )
+
+    def _update_version_files(self, new_version) -> List[Path]:
+        config = self._load_config()
+
+        changed_files: List[Path] = []
+        for file_config in config["files"]:
+            file_path = file_config["path"]
+            with open(Path.cwd() / file_path, "r") as input_file_handle:
+                lines = input_file_handle.readlines()
+                line_number = file_config["line"]
+                version_line = lines[line_number - 1]
+
+                matches = re.match(VERSION_PATTERN, version_line, re.DOTALL)
+                if matches is None:
+                    raise VersionError(
+                        f"Line has no version, "
+                        f"file:'{file_path}' "
+                        f"lineNo:{line_number} "
+                        f"content:'{version_line}'"
+                    )
+                lines[line_number - 1] = (
+                    matches.group("pre")
+                    + str(new_version)
+                    + matches.group("post")
+                )
+
+                content = "".join(lines)
+                with open(Path.cwd() / file_path, "w") as output_file_handle:
+                    output_file_handle.write(content)
+                changed_files.append(Path(file_config["path"]))
+        return changed_files
+
+    def _load_config(self) -> Dict:
```

Comment:
```suggestion
    def _load_config(self) -> Dict[str, Any]:
```
Maybe the dict can be typed even more precise like `Dict[str, Union[int, str]]`?